### PR TITLE
Tests for error cases

### PR DESCRIPTION
**Pull request by aaronpowell**
This aims to get #37 covered as well as we can

### CODE REVIEW

**Comment by aaronpowell** [comment]
I think this as close to 100% as is really practical to go, there's a few tweaks that could be done to get it a bit closer but it'd require the tests to become more complex and I'm in favour of having less complex tests at the expense of a bit of coverage.

File: nvm.tests.ps1 [hunk 1]
```diff
@@ -190,6 +190,31 @@ Describe "Set-NodeVersion" {
...
+                    return @{
+                        engines = @{
+                        }
+                    } | ConvertTo-Json
+                }
+
+                { Set-NodeVersion } | Should Throw
+            }
+
+            It "Will error is no version, no .nvmrc and now package.json" {
```
Comment (felixfbecker) [inline nvm.tests.ps1]: Two typos

File: nvm.psm1 [hunk 2]
```diff
@@ -109,11 +109,6 @@ function Set-NodeVersion {
         $requestedVersion
     }
 
-    if (!(Test-Path -Path $requestedVersion)) {
-        "Could not find node version $matchedVersion"
-        return
-    }
```
Comment (felixfbecker) [inline nvm.psm1:115]: What’s the reason to remove this?

Comment (aaronpowell) [reply nvm.psm1:115]: Have a read of the commit message for 2323f82049a6755820f29bf75ad5a2e38d322615, it explains it

**Review by felixfbecker — APPROVED** [review]
Looking at the coverage report I think there are still a few more error cases that could be tested, but the current coverage % is pretty good for such a short script.

**Pull request merged**